### ISSUE TITLE
Referencing deployments within deployments with relativePath is not documented well

### ISSUE THREAD

**Issue by slavizh** [Azure/template-specs#18]
I have deployment like:
- azuredeploy.json - the main template file
- template1.json - called from azuredeploy.json. Located in nestedtemplates folder
- template2.json  - called from template1.json.  Located in nestedtemplates folder

When you have to specify relativePath for template1.json in azuredeploy.json it looks like this:
nestedtemplates/template1.json

When I have to specify relativePath for template2 in template1.json it looks like this:
template2.json

Basically the relativePath value needs to be accordance to the path of the template which is the parent. This could be confusing. For me best approach relative path to always be from the root of your solution structure.
